File: RiskProducer/RiskMessage.cs
using System;
using System.Collections.Generic;

namespace RiskProducer
{
    public class RiskMessage
    {
        public RiskMessage(Guid tradeId, decimal amount, int version, DateTime createdAt, Hierarchy hierarchy)
        {
            TradeId = tradeId;
            Amount = amount;
            Version = version;
            CreatedAt = createdAt;
            Hierarchy = hierarchy;
        }

        public Guid TradeId { get; }
        public decimal Amount { get; }
        public int Version { get; }
        public DateTime CreatedAt { get; }
        public Hierarchy Hierarchy { get; }
    }
}

File: acceptance-test/domain/risk-message.ts
export class RiskMessage {
    public TradeId: string;
    Amount: number;
    Version: number;
    CreatedAt: Date;
    Hierarchy: Hierarchy;

    constructor(
        tradeId: string, 
        amount: number, 
        version: number, 
        createdAt: Date, 
        hierarchy: Hierarchy) {
        this.TradeId = tradeId;
        this.Amount = amount;
        this.Version = version;
        this.CreatedAt = createdAt;
        this.Hierarchy = hierarchy;
    }
}

export class Hierarchy {
    RiskType: string;
    TradeDesk: string;
    Region: string;

    constructor(riskType: string, tradeDesk: string, region: string) {
        this.RiskType = riskType;
        this.TradeDesk = tradeDesk;
        this.Region = region;
    }
}

export interface IRiskMessageInputRow {
    Amount: string;
    Hierarchy: string; 
}

export class RiskMessageInputRow {
    public Amount: number;
    public Hierarchy: Hierarchy;

    constructor({Amount, Hierarchy: HierarchyId}: IRiskMessageInputRow) {
        this.Amount = Number(Amount);

        var hierarchyList = HierarchyId.split(":", 3);
        this.Hierarchy = new Hierarchy(hierarchyList[0], hierarchyList[1], hierarchyList[2]);
    }
}

// export interface IRiskMessage {
//     TradeId: string;
//     Amount: number;
//     Version: number;
//     CreatedAt: Date;
//     Hierarchy: IHierarchy;
// }

// export interface IHierarchy{
//     RiskType: string;
//     TradeDesk: string;
//     Region: string;
// }